src/ispd/gui/UserConfigurationDialog.java

```diff
@@ -149,18 +150,23 @@ private void onAddClick(final ActionEvent evt) {
         );
 
         // TODO: Parse result and error.
-        final Double result =
-                Double.parseDouble(JOptionPane.showInputDialog(this,
-                "Enter user power comsumption limit"));
+        final String sResult =JOptionPane.showInputDialog(this,
+                        "Enter user power comsumption limit");
+
 
-        if (!this.userSet.contains(add) && !add.isEmpty()) {
+        if (!add.isEmpty() && !this.userSet.contains(add)  && !sResult.isEmpty() && !this.userSet.contains(sResult) ){
+            final Double result = Double.parseDouble(sResult);
```

Comment:
Test when the `sResult` is not a valid double, should be enclosed in a try/catch block.